main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+  "net/http"
+  "github.com/gin-gonic/gin"
+)
+
+type album struct {
+  ID     string `json:"id"`
+  Title  string `json:"title"`
+  Artist string `json:"artist"`
+  Price  float64 `json:"price"`
+}
+
+var albums = []album{
```

Comment:
For this example as it currently is, it's probably OK, but global variables like this are strongly discouraged -- especially if they can change at run time like this one can (it grows in `postAlbums`).

Learning why this is bad requires you to first experience feeling the pain of this kind of code over time, so I actually recommend writing *more* code around this anti-pattern until you can appreciate and verbalise why it's bad.

The best way is probably to try writing multiple tests for your code. It's a big challenge, as you'll have to learn how to write tests against HTTP endpoints as well, but the learning experience should be good. We can go over it 1-1 if/after you struggle.

Reply:
Will be addressed after working on unit testing.